src/line.rs

```diff
@@ -267,6 +267,26 @@ impl<'input, 'header, Endian> StateMachine<'input, 'header, Endian>
             }
         }
     }
+
+    /// Parse and execute opcodes until we reach a row matching `addr`, the end of the program,
+    /// or an error.
+    pub fn run_to_address(&mut self, addr: &u64)
+                          -> parser::ParseResult<Option<&LineNumberRow<'input, 'header, Endian>>> {
+        loop {
+            match self.next_row() {
+                Ok(Some(row)) => {
+                    if row.address() == *addr {
+                        // Can't return 'row' directly here because of rust-lang/rust#21906.
```

Comment:
Yep. Non-lexical borrows will help a lot when they come, but in the meantime you'll get used to restructuring your programs to avoid these kinds of issues / play better with the borrow checker, and only need these weird control flows when absolutely necessary. I don't really have anything concrete to say, just an intuition that all the higher order functions and iterator combinators and things help with this stuff.